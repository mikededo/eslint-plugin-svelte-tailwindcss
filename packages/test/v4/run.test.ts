import { spawnSync } from "node:child_process";

describe('e2e', () => {
  it.each(['default', 'config', 'settings'] as const)('should lint with the %s configuration', async (type) => {
    const result = spawnSync(`TEST_TYPE=${type === 'default' ? '' : type} bun lint`, [], {
      encoding: 'utf-8',
      shell: true 
    });

    expect(result.stdout).toMatchSnapshot('stdout');
    expect(result.status).toMatchSnapshot('exit code');
  });
})
