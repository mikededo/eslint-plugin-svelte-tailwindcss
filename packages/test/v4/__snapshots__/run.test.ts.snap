// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e > should lint with the config configuration > exit code 1`] = `1`;

exports[`e2e > should lint with the config configuration > stdout 1`] = `
"
/v4
> eslint . -c eslint.config.test.mjs


/v4/src/components/button/styles.ts
  39:3  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/components/button/text-icon-button.svelte
  19:9  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/components/checkbox/checkbox.svelte
  15:9  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/components/container.svelte
   8:6  error    Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
   8:6  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  12:1  warning  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

/v4/src/routes/+layout.svelte
  7:6  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/routes/+page.svelte
   4:20  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  17:6   error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  18:6   error    Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
  20:1   warning  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

/v4/src/routes/styles.ts
   3:7  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes
  10:1  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

✖ 16 problems (14 errors, 2 warnings)
  14 errors and 2 warnings potentially fixable with the \`--fix\` option.

 ELIFECYCLE  Command failed with exit code 1.
"
`;

exports[`e2e > should lint with the default configuration > exit code 1`] = `1`;

exports[`e2e > should lint with the default configuration > stdout 1`] = `
"
/v4
> eslint . -c eslint.config.test.mjs


/v4/src/components/container.svelte
   8:6  error  Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
   8:6  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  12:1  error  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

/v4/src/routes/+layout.svelte
  7:6  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/routes/+page.svelte
  17:6  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  18:6  error  Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
  20:1  error  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

✖ 7 problems (7 errors, 0 warnings)
  7 errors and 0 warnings potentially fixable with the \`--fix\` option.

 ELIFECYCLE  Command failed with exit code 1.
"
`;

exports[`e2e > should lint with the settings configuration > exit code 1`] = `1`;

exports[`e2e > should lint with the settings configuration > stdout 1`] = `
"
/v4
> eslint . -c eslint.config.test.mjs


/v4/src/components/button/styles.ts
  39:3  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/components/button/text-icon-button.svelte
  19:9  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/components/checkbox/checkbox.svelte
  15:9  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/components/container.svelte
   8:6  error  Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
   8:6  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  12:1  error  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

/v4/src/routes/+layout.svelte
  7:6  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

/v4/src/routes/+page.svelte
   4:20  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  17:6   error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  18:6   error  Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
  20:1   error  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

/v4/src/routes/styles.ts
   3:7  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes
  10:1  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

✖ 16 problems (16 errors, 0 warnings)
  16 errors and 0 warnings potentially fixable with the \`--fix\` option.

 ELIFECYCLE  Command failed with exit code 1.
"
`;
